packages/grant-explorer/src/features/contributors/ViewContributionHistory.tsx

```diff
@@ -0,0 +1,395 @@
+      data: { chainId: number; data: Contribution[] | never[] }[];
+    }
+  | { type: "error"; error: string };
+
+const useContributionHistory = (chainIds: number[], rawaddress: string) => {
+  const [state, setState] = useState<ContributionHistoryState>({
+    type: "loading",
+  });
+
+  useEffect(() => {
+    console.log("loading", chainIds, rawaddress);
+
+    if (!process.env.REACT_APP_ALLO_API_URL) {
+      throw new Error("REACT_APP_ALLO_API_URL is not set");
+    }
+
+    const histories: { chainId: number; data: Contribution[] | never[] }[] = [];
+    let allChainsError = true;
+
+    const fetchContributions = async () => {
+      const fetchPromises = chainIds.map((chainId: number) => {
+        if (!process.env.REACT_APP_ALLO_API_URL) {
+          throw new Error("REACT_APP_ALLO_API_URL is not set");
+        }
+
+        const client = new AlloIndexerClient(
+          fetch.bind(window),
+          process.env.REACT_APP_ALLO_API_URL,
+          chainId
+        );
+
+        let address = "";
+        try {
+          // ensure the address is a valid address
+          address = ethers.utils.getAddress(rawaddress.toLowerCase());
+        } catch (e) {
+          setState({
+            type: "error",
+            error: "bad address",
+          });
+        }
+
+        return client
+          .getContributionsByAddress(address)
+          .then((data) => {
+            const history = { chainId, data };
+            histories.push(history);
+            allChainsError = false; // If data is fetched successfully for any chain, set allChainsError to false
+          })
+          .catch((error) => {
+            console.log(
+              `Error fetching contribution history for chain ${chainId}:`,
+              error
+            );
+            const history = { chainId, data: [] };
+            histories.push(history);
+          });
+      });
+
+      await Promise.all(fetchPromises);
+
+      if (allChainsError) {
+        setState({
+          type: "error",
+          error: "Error fetching contribution history for all chains",
+        });
+      } else {
+        console.log("loaded", histories);
+        setState({ type: "loaded", data: histories });
+      }
+    };
+
+    fetchContributions();
+  }, [chainIds, rawaddress]);
+
+  return state;
+};
+
+function StatCard(props: { title: string; value: string | undefined }) {
+  return (
+    <div className="rounded border border-violet-400 p-4 flex-grow">
+      <div className="font-bold text-md pb-4">{props.title}</div>
+      <div className="text-grey-400 text-2xl">{props.value}</div>
+    </div>
+  );
+}
+
+export function ViewContributionHistoryDisplay(props: {
+  tokens: Record<string, PayoutToken>;
+  contributions: { chainId: number; data: Contribution[] | never[] }[];
+  address: string;
+}) {
+  const { data: ensName } = useEnsName({
+    address: props.address,
+  });
+  const addressLogo = blockies
+    .create({ seed: props.address.toLowerCase() })
```

Comment:
this is generating an image which we should not do on render, we should do it in a `useMemo`